executor/linux/build.go

```diff
@@ -520,6 +529,61 @@ func (c *client) ExecBuild(ctx context.Context) error {
 			continue
 		}
 
+		// iterate through each secret provided in the pipeline
+		for _, secret := range c.pipeline.Secrets {
+			_log, err := step.LoadLogs(c.init, &c.stepLogs)
+			if err != nil {
+				return err
+			}
+
+			_log.SetData([]byte(""))
+
+			// ignore pulling secrets coming from plugins
+			if !secret.Origin.Empty() {
+				continue
+			}
+
+			c.Logger.Infof("pulling secret %s", secret.Name)
+			// only pull in secrets that are set to be pulled in at the start
+			if strings.EqualFold(secret.Pull, constants.SecretPullBuild) {
+				continue
+			}
+
+			//nolint:contextcheck // ignore passing context
+			s, err := c.secret.pull(secret)
+			if err != nil {
+				c.err = err
+				return fmt.Errorf("unable to pull secrets: %w", err)
+			}
+
+			_log.AppendData([]byte(
+				fmt.Sprintf("$ vela view secret --secret.engine %s --secret.type %s --org %s --repo %s --name %s \n",
+					secret.Engine, secret.Type, s.GetOrg(), s.GetRepo(), s.GetName())))
+
+			sRaw, err := json.MarshalIndent(s.Sanitize(), "", " ")
+			if err != nil {
+				c.err = err
+				return fmt.Errorf("unable to decode secret: %w", err)
+			}
+
+			_log.AppendData([]byte(string(sRaw)[strings.Index(string(sRaw), "name")-2 : strings.Index(string(sRaw), "value")-2]))
+			_log.AppendData([]byte(string(sRaw)[strings.Index(string(sRaw), "images")-2 : strings.Index(string(sRaw), "created_at")-2]))
```

Comment:
can we add some index bounds checks here?

Reply:
I could, but I figured it would be good because I'm pretty sure every sRaw includes name, value, images, and created_at? But if that's still a concern I can add that

Reply:
yeah i suppose youre right. but worst case scenario... the worker will crash :(

Reply:
Oof ok yeah will add!